modules/adspendBidAdapter.js

```diff
@@ -0,0 +1,120 @@
+import * as utils from 'src/utils';
+import { config } from 'src/config';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER, NATIVE, VIDEO } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'adspend';
+const URL = '//rtb.com.ru/headerbidding-bid';
+const COOKIE_NAME = 'hb-adspend-id';
+const UUID_LEN = 36;
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['adspend'],
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+
+  isBidRequestValid: function(bid) {
+    const conf = config.getConfig();
+    return !!(conf.currency && conf.currency.adServerCurrency && bid.crumbs.pubcid);
+  },
+
+  buildRequests: function(bidRequests, bidderRequest) {
+    const req = bidRequests[Math.floor(Math.random() * bidRequests.length)];
+
+    const conf = config.getConfig();
+    const cur = conf.currency.adServerCurrency;
+
+    const bidfloor = req.params.bidfloor;
+
+    const payload = {
+      'id': req.bidId,
+      'site': {
+        'id': req.crumbs.pubcid,
+        'domain': document.domain
+      },
+      'device': {
+        'ua': navigator.userAgent,
+        'ip': ''
+      },
+      'user': { 'id': getUserID() },
+      'imp': [
+        {
+          'id': req.params.placementId,
+          'tagid': req.params.tagId,
+          'banner': { 'format': getFormats(req.sizes) },
+          'bidfloor': bidfloor !== undefined ? Number(bidfloor) : 1,
+          'bidfloorcur': cur,
+          'secure': 0
+        }
+      ],
+      'cur': [
+        cur
+      ],
+      'tmax': bidderRequest.timeout
+    };
+
+    return {
+      method: 'POST',
+      url: URL,
+      data: JSON.stringify(payload),
+    };
+  },
+
+  interpretResponse: function(resp, {bidderRequest}) {
+    const bids = [];
+
+    if (resp.body === '') return bids;
+
+    const respBid = resp.body.seatbid[0].bid[0];
+    const bid = {
+      cpm: respBid.price,
+      requestId: resp.body.id,
+      width: 300,
+      height: 250,
```

Comment:
Should these values be hard-coded?  Wouldn't the size dimensions be part of the response?

Reply:
`width` and `height` shouldn't. I'll fix it